common-npm-packages/docker-common-v2/registryauthenticationprovider/registryauthenticationtoken.ts

```diff
@@ -55,13 +55,23 @@ export default class RegistryServerAuthenticationToken {
     }
 }
 
-export function getDockerRegistryEndpointAuthenticationToken(endpointId: string): RegistryServerAuthenticationToken {
+export async function getDockerRegistryEndpointAuthenticationToken(endpointId: string): Promise<RegistryServerAuthenticationToken> {
     var registryType = tl.getEndpointDataParameter(endpointId, "registrytype", true);
     let authToken: RegistryServerAuthenticationToken;
 
     if (registryType === "ACR") {
         const loginServer = tl.getEndpointAuthorizationParameter(endpointId, "loginServer", false).toLowerCase();;
-        authToken = new ACRAuthenticationTokenProvider(endpointId, loginServer).getAuthenticationToken();
+        let authType: string;
+        try {
+            authType = tl.getEndpointAuthorizationScheme(endpointId, false);
+        } catch {
+            authType = tl.getEndpointAuthorizationParameter(endpointId, "scheme", false);
+        }
+        if (authType === "ManagedServiceIdentity") {
+            authToken = await new ACRAuthenticationTokenProvider(endpointId, loginServer).getMSIAuthenticationToken(0, 0);
```

Comment:
Readability: What does (0,0) signifies here ? Can you please label the parameters or add code comments ?